GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -330,30 +356,97 @@ - (instancetype)init {
   return self;
 }
 
+/** Populates a unique NSUInteger as GDTCORBackgroundIdentifier for given semaphore and maps it to
+ * the semaphore in NSMutableDictionary.
+ *
+ * @param semaphore The task semaphore needs to be mapping.
+ * @return A unique GDTCORBackgroundIdentifier mapped to the given semaphore.
+ */
++ (GDTCORBackgroundIdentifier)createAndMapBackgroundIdentifierToSemaphore:
+    (dispatch_semaphore_t)semaphore {
+  __block GDTCORBackgroundIdentifier bgID = GDTCORBackgroundIdentifierInvalid;
+  if (sSemaphoreQueue && sBackgroundIdentifierToSemaphoreMap) {
+    dispatch_sync(sSemaphoreQueue, ^{
```

Comment:
It's technically possible for sSemaphoreQueue and sBackgroundIdentifierToSemaphoreMap to be deallocated between lines 368 and 369. The way you solve this is you read them into local variables (which will have a strong reference until the end of this method) and then check the values of the variables. That ensures that they can't be deallocated after you checked but before you used them.